public/app/features/dashboard-scene/scene/layouts-shared/utils.ts

```diff
@@ -34,3 +34,35 @@ export function useEditPaneInputAutoFocus({ autoFocus }: EditPaneInputAutoFocusP
 
   return ref;
 }
+
+export function generateUniqueTitle(title: string | undefined, existingTitles: Set<string>): string {
```

Comment:
I'd go for something simpler here:

```ts
function getNextNumberedString(base: string, existing: string[]): string {
  const regexPrefix = new RegExp(`^${base}\\s(\\d+)\$`);

  const maxNumber = existing.reduce<number>((acc,str) => {
    const match = str.match(regexPrefix);

    if (!match || match.length < 2) {
      return acc;
    }

    const number = parseInt(match[1], 10);

    if (isNaN(number)) {
      return acc;
    }

    return Math.max(acc, number);
  }, 0);

  return `${base} ${maxNumber + 1}`;
}
```

Basically we go through the existing titles and find the max number then we return a title from the base text and max number + 1.

Reply:
This only works for new rows though. If I paste a row called `Row 1` into a manager that has `Row 1` and `Row 2`, the name of the pasted row will be `Row 1 1`

Reply:
Argh :-/. Yeah, that complicates things quite a bit.